src/main/scala/org/hammerlab/guacamole/reads/Read.scala

```diff
@@ -61,14 +61,14 @@ trait Read {
   val isDuplicate: Boolean
 
   /** Is this read mapped? */
-  final val isMapped: Boolean = getMappedReadOpt.isDefined
+  final val isMapped: Boolean = this match {
```

Comment:
semi-related thought: should we `@transient` some/all fields in classes that get ser'd on the critical path, e.g. `Read`, `Pileup`?